test/tests/performance/pbs_cgroups_stress.py

```diff
@@ -0,0 +1,233 @@
+            for line in fd:
+                entries = line.split()
+                if entries[2] != "cgroup":
+                    continue
+                flags = entries[3].split(',')
+                subsys = os.path.basename(entries[1])
+                paths[subsys] = entries[1]
+                if 'memory' in flags:
+                    paths['memsw'] = paths[subsys]
+                    paths['memory'] = paths[subsys]
+        return paths
+
+    def load_hook(self, filename):
+        """
+        Import and enable a hook pointed to by the URL specified.
+        """
+        try:
+            with open(filename, 'r') as fd:
+                script = fd.read()
+        except IOError:
+            self.assertTrue(False, "Failed to open hook file %s" % filename)
+        events = '"execjob_begin,execjob_launch,execjob_attach,'
+        events += 'execjob_epilogue,execjob_end,exechost_startup,'
+        events += 'exechost_periodic"'
+        a = {'enabled': 'True',
+             'freq': '2',
+             'event': events}
+        self.server.create_import_hook(self.hook_name, a, script,
+                                       overwrite=True)
+        # Add the configuration
+        self.load_config(self.cfg0)
+
+    def load_config(self, cfg):
+        """
+        Create a hook configuration file with the provided contents.
+        """
+        (fd, fn) = self.du.mkstemp()
+        os.write(fd, cfg)
+        os.close(fd)
+        a = {'content-type': 'application/x-config',
+             'content-encoding': 'default',
+             'input-file': fn}
+        self.server.manager(MGR_CMD_IMPORT, HOOK, a, self.hook_name)
+        os.remove(fn)
+        self.mom.log_match('pbs_cgroups.CF;copy hook-related ' +
+                           'file request received',
+                           max_attempts=5,
+                           starttime=self.server.ctime)
+        self.logger.info("Current config: %s" % cfg)
+        # Restart MoM to work around PP-993
+        self.mom.restart()
+
+    @timeout(600)
+    def test_cgroups_race_condition(self):
+        """
+        Test to ensure a cgroups event does not read the cgroups file system
+        while another event is writing to it. By submitting 1000 instant jobs,
+        the events should collide at least once.
+        """
+        pcpus = 0
+        with open('/proc/cpuinfo', 'r') as desc:
+            for line in desc:
+                if re.match('^processor', line):
+                    pcpus += 1
+        if pcpus < 8:
+            self.skipTest("Test requires at least 8 physical CPUs")
+
+        attr = {'job_history_enable': 'true'}
+        self.server.manager(MGR_CMD_SET, SERVER, attr)
+        self.load_config(self.cfg0)
+        now = time.time()
+        j = Job(TEST_USER, attrs={ATTR_J: '0-1000'})
+        j.create_script(self.true_script)
+        jid = self.server.submit(j)
+        jid = jid.split(']')[0]
```

Comment:
Sped up the string building just a tiny bit by splitting on the close brace, then only storing the first string. An example:
```Python
jid = submit               # jid = "101[]"
jid = jid.split(']')[0]    # jid = "101["
sjid = jid + str(i) + "]"  # sjid = "101[1]"
```